### PR TITLE
[BugFix] Failed delete job do not return until timeout

### LINKED ISSUE #2527: Wrong result return while date type of column is `datetime`.
insert and delete error.
1. Precision problem of `datetime` when insert into table values.
2. Delete error('ERROR 1064 (HY000): Invalid column value[null] for column k2')
### Steps to reproduce the behavior (Required)
```
mysql> CREATE TABLE test_string2 (`k1` string not null,k2 datetime) ENGINE=OLAP duplicate key(k1) DISTRIBUTED BY HASH(`k1`) BUCKETS 3 PROPERTIES ("replication_num" = "1");
Query OK, 0 rows affected (0.01 sec)

mysql> insert into test_string2 values('2021-12-29 16:58:56.032000','2021-12-29 16:58:56.032000');
Query OK, 1 row affected (0.03 sec)
{'label':'insert_5e9563d2-689f-11ec-8fab-00163e10c9d6', 'status':'VISIBLE', 'txnId':'33055'}

mysql> select * from test_string2;
+----------------------------+----------------------------+
| k1                         | k2                         |
+----------------------------+----------------------------+
| 2021-12-29 16:58:56.032000 | 2021-12-29 16:58:56.032000 |
+----------------------------+----------------------------+
1 row in set (0.01 sec)

mysql> delete from test_string2 where k2='2021-12-29 16:58:56.032000';
ERROR 1064 (HY000): Invalid column value[null] for column k2
mysql> delete from test_string2 where k1='2021-12-29 16:58:56.032000';
Query OK, 0 rows affected (0.02 sec)
{'label':'delete_a088fb14-0748-44ce-b354-18e3b66fc20d', 'status':'VISIBLE', 'txnId':'33056'}

mysql> select * from test_string2;
Empty set (0.00 sec)
```

-mysql
```
mysql>  CREATE TABLE test_string2 (`k1` datetime not null,k2 datetime);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into test_string2 values('2021-12-29 16:58:56.032000','2021-12-29 16:58:56.032000');
Query OK, 1 row affected (0.00 sec)

mysql> select * from test_string2;
+---------------------+---------------------+
| k1                  | k2                  |
+---------------------+---------------------+
| 2021-12-29 16:58:56 | 2021-12-29 16:58:56 |
+---------------------+---------------------+
1 row in set (0.00 sec)
```

### Expected behavior (Required)
```
mysql> select * from test_string2;
+---------------------+---------------------+
| k1                  | k2                  |
+---------------------+---------------------+
| 2021-12-29 16:58:56 | 2021-12-29 16:58:56 |
+---------------------+---------------------+
1 row in set (0.00 sec)
```
### Real behavior (Required)
```
mysql> select * from test_string2;
+----------------------------+----------------------------+
| k1                         | k2                         |
+----------------------------+----------------------------+
| 2021-12-29 16:58:56.032000 | 2021-12-29 16:58:56.032000 |
+----------------------------+----------------------------+
1 row in set (0.01 sec)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - 2.1

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes https://github.com/StarRocks/starrocks/issues/2527

## Problem Summary(Required):

If a delete job execute failed in BE and it return failed to FE, FE will ignore the fail status and retry until timeout. It will confuse users and ignore the real fail reason.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 60 (45.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/MarkedCountDownLatch.java|0|6|00.00%|[68, 69, 71, 72, 73, 75]|
|:large_blue_circle:|com/starrocks/task/PushTask.java|0|5|00.00%|[225, 226, 227, 228, 231]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|7|00.00%|[227, 238, 239, 240, 241, 242, 244]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|0|1|00.00%|[483]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|27|41|65.85%|[229, 230, 231, 241, 259, 260, 261, 262, 263, 264, 265, 266, 267, 269]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 59 (45.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/MarkedCountDownLatch.java|0|6|00.00%|[68, 69, 71, 72, 73, 75]|
|:large_blue_circle:|com/starrocks/task/PushTask.java|0|5|00.00%|[225, 226, 227, 228, 231]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|7|00.00%|[227, 238, 239, 240, 241, 242, 244]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|27|41|65.85%|[229, 230, 231, 241, 259, 260, 261, 262, 263, 264, 265, 266, 267, 269]|

**Review by Astralidea — APPROVED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 61 (44.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/MarkedCountDownLatch.java|0|6|00.00%|[68, 69, 71, 72, 73, 75]|
|:large_blue_circle:|com/starrocks/task/PushTask.java|0|5|00.00%|[225, 226, 227, 228, 231]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|9|00.00%|[227, 238, 239, 240, 241, 242, 244, 245, 246]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|27|41|65.85%|[229, 230, 231, 241, 259, 260, 261, 262, 263, 264, 265, 266, 267, 269]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/MarkedCountDownLatch.java [hunk 1]
```diff
@@ -64,6 +64,17 @@ public synchronized boolean markedCountDown(K key, V value) {
         return false;
     }
 
+    public synchronized boolean markedCountDown(K key, V value, Status status) {
+        if (st.ok()) {
+            st = status;
+        }
+        if (marks.remove(key, value)) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/MarkedCountDownLatch.java:71]: you can reuse `markedCountDown(key, value)` here.

Comment (sevev) [reply fe/fe-core/src/main/java/com/starrocks/common/MarkedCountDownLatch.java]: > you can reuse `markedCountDown(key, value)` here.

done, changed in https://github.com/StarRocks/starrocks/pull/23320/commits/1836a3a138a9ea19ff8795e0315bf04e1373a2d1

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 58 (46.55%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/MarkedCountDownLatch.java|0|3|00.00%|[68, 69, 71]|
|:large_blue_circle:|com/starrocks/task/PushTask.java|0|5|00.00%|[225, 226, 227, 228, 231]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|9|00.00%|[227, 238, 239, 240, 241, 242, 244, 245, 246]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|27|41|65.85%|[229, 230, 231, 241, 259, 260, 261, 262, 263, 264, 265, 266, 267, 269]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 58 (46.55%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/MarkedCountDownLatch.java|0|3|00.00%|[68, 69, 71]|
|:large_blue_circle:|com/starrocks/task/PushTask.java|0|5|00.00%|[225, 226, 227, 228, 231]|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|9|00.00%|[227, 238, 239, 240, 241, 242, 244, 245, 246]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|27|41|65.85%|[229, 230, 231, 241, 259, 260, 261, 262, 263, 264, 265, 266, 267, 269]|

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by yongbingwang** [comment]
branch-2.5
https://github.com/StarRocks/starrocks/pull/26123

**Comment by yongbingwang** [comment]
branch-3.0
https://github.com/StarRocks/starrocks/pull/26122

**Pull request merged**